src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    private static final Random random = new Random();
+
+    public static Ball getRandomBall() {
+        Color randomColor = ColorSupplier.getRandomColor();
+        int randomNumber = random.nextInt(100) + 1;
+        return new Ball(randomColor, randomNumber);
+    }
+
+}
```

Comment:
The usage of inheritance here (`Lottery extends ColorSupplier`) is not justified because there's no 'is-a' relationship. Lottery is not a type of ColorSupplier. Instead, you should use composition over inheritance and have a ColorSupplier instance within the Lottery class.